### PR TITLE
Validate more parameters in the config file

### PR DESCRIPTION
**Pull request by CyberDem0n**
- parameters for different DCS
- more bootstrap.dcs parameters
- ctl, restapi, and watchdog parameters

### CODE REVIEW

File: patroni/validator.py [hunk 1]
```diff
@@ -827,31 +850,99 @@ def validate_watchdog_mode(value: Any) -> None:
         "srv": str,
         "srv_suffix": str,
         "url": str,
-        "proxy": str})
+        "proxy": str
+    }),
+    Optional("protocol"): str,
+    Optional("username"): str,
```
Comment (hughcapet) [inline patroni/validator.py:856]: what's about password?

File: patroni/validator.py [hunk 2]
```diff
@@ -827,31 +850,99 @@ def validate_watchdog_mode(value: Any) -> None:
...
-                "url": str})
+                "url": str
+            }),
+            Optional("port"): int,
+            Optional("scheme"): str,
+            Optional("token"): str,
+            Optional("verify"): bool,
+            Optional("cacert"): str,
+            Optional("cert"): str,
+            Optional("key"): str
```
Comment (hughcapet) [inline patroni/validator.py:945]: and dc, consistency, checks...?

File: patroni/validator.py [hunk 3]
```diff
@@ -802,6 +803,28 @@ def __call__(self, value: Union[int, str]) -> bool:
         return ret
 
 
+class EnumValidator(object):
+    """Validate enum setting
+
+    :ivar allowed_values: a tuple with allowed enum values.
```
Comment (hughcapet) [inline patroni/validator.py:809]: I guess, it is not tuple

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**